docs/development/pigs/pig-010.rst

```diff
@@ -0,0 +1,460 @@
+also includes non-spatial regions, e.g. spectral regions or time regions.
+
+Just to mention what we have at the moment: spectral "regions" are handled via
+``energy_bounds`` or ``(energy_min, energy_max)`` in a not very consistent way.
+We could look towards the Astropy `specutils`_ package and the
+`specutils.SpectralRegion`_ object as an example how to implement spectral
+regions. Currently there is plan to integrate ``astropy-regions`` and
+``specutils``, `astropy-regions GH 228`_ has been opened for discussion.
+
+Supporting an energy-dependent maximum field of view radius selection, or a
+field-of-view dependent safe energy offset might be a motivation to couple
+spatial and spectral region, possibly supported via ``gammapy.maps``.
+
+For time regions we currently often have ``(time_min, time_max)`` in the API,
+and we have the `gammapy.data.GTI`_ class to represent multiple intervals. It's
+not clear if this needs to be changed or wrapped somehow into a "time region",
+or if there are any use cases to couple time with the spatial and spectral
+regions, e.g. for a time-dependent safe energy threshold or user analysis
+option.
+
+These thoughts are related to the question in `GH 1805`_: "Introduce RegionGeom
+and RegionNDMap?" - Axel: do you want this for v1.0 or postpone? If v1.0, then
+do you want to flesh out a proposal here and now, or later in a separate PIG?
+
+In the HESS software, there is a region class that wraps a mask, and that is
+sometimes useful, specifically because in the HESS software exclusion regions
+are region objects, sometimes compound with a mask region and other regions like
+circles. It's clearly an option, easy to implement as a ``SkyRegion`` sub-class.
+But also causes problems, especially when it comes to serialisation. Thoughts?
+I'm OK to not add this for now, and just use ``Map`` objects for exclusion
+masks, and require that users create that before using Gammapy, in case they
+e.g. have a list of circles or whatever they want to use as exclusion region.
+
+Task list
+=========
+
+We propose to implement this PIG though a series of small pull requests. Some are
+dependent on others, but for many also the order doesn't matter and overall the
+amount of work is limited.
+
+- Add ``PixRegion.rotate(center, angle)`` in ``astropy-healpix``. See `Rotated regions`_ above.
+- Release ``astropy-healpix`` v0.4, then bump Gammapy dependency to that version.
+- Improve rotated region code and tests and docs in Gammapy.
+  This could mean a rebase and continuation of `GH 2089`_, or closing that and
+  re-starting from master, copying over the useful parts or cherry-picking
+  useful commits from that branch.
+- Add ``gammapy/utils/regions.py`` with two helper functions. See `Region arguments`_ above.
+- Use the helper functions throughout Gammapy. Should be 5-10 cases, could be
+  several PRs. See `Region arguments`_ above.
+- Replace sky circle and box select with generic region select (see `Sky and
```

Comment:
It's easier to first add the `region_select` and then to remove the old methods, because then one can just migrate callers and tests to that. And I wanted to put the work in `astropy-regions` first, because it's the basis and we need that in a stable release before we can use it in Gammapy. So I'd prefer to keep order of tasks as-is (but really all of that work is only about a week full-time for this PIG, should all happen soon).

Reply:
I've clarified that these changes in Gammapy are independent of the ones with `region.rotate`, could happen in any order.